### ISSUE TITLE
p-calendar Month Navigator doesn't reflect the currently displayed month and always shows 'January'

### ISSUE THREAD

**Issue by fijas** [primefaces/primeng#6094]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
https://stackblitz.com/edit/github-akxcws

**Current behavior**
On changing the month using month navigator, the calendar changes but the month navigator does not reflect the new month. It always only shows 'January'. This bug can also be found in the PrimeNG documentation at: https://www.primefaces.org/primeng/#/calendar under the navigation demo.

**Expected behavior**
The month navigator should reflect the currently displayed month, but it doesn't.

**Minimal reproduction of the problem with instructions**
https://stackblitz.com/edit/github-akxcws
Try using the month navigator to change months in the calendar

**What is the motivation / use case for changing the behavior?**
Not an expected behavior

**Please tell us about your environment:**
All OSes

* **Angular version:** 5.X
6.0.9

* **PrimeNG version:** 5.X
6.0.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Latest Chrome and Firefox

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript 2.7.2

* **Node (for AoT issues):** `node --version` =

**Comment by tsupakit** [comment]
This seems fixed on v6.0.2
But found another case which is combining navigation and button bar.
When change navigation month or year and click Today button, the selection does not change to current month and year.

**Comment by cagataycivici** [comment]
https://github.com/primefaces/primeng/issues/6261
